pkg/model/user.go

```diff
@@ -38,6 +39,18 @@ func (u *User) Create(db *sql.DB) error {
 	return db.QueryRow("INSERT INTO auth.users(id, email, password, confirmToken) VALUES($1, $2, $3, $4) RETURNING id", u.ID, u.Email, u.Password, u.ConfirmToken).Scan(&u.ID)
 }
 
+// GeneratePassword generate random password
+func (u *User) GeneratePassword(length int) string {
```

Comment:
I think that your can rename it to `CreateRandomPassword` and mutate the user instance directly inside this function.
Then, on your handler you can call `user.CreateRandomPassword(12)`. 
Also delete the password creation on the providers methods.

Reply:
And then, you can directly call the `user.HashPassword()` inside this method.